src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -135,13 +141,48 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter);
         }
     };
 }
 
 using namespace RSS;
 
+QString computeEpisodeName(const QString &article)
+{
+    QRegularExpression episodeRegex = AutoDownloader::instance()->getSmartEpisodeRegex();
+    QRegularExpressionMatch match = episodeRegex.match(article);
+
+    // See if we can extract an season/episode number or date from the title
+    if (!match.hasMatch()) {
+        return QString();
+    }
+
+    QStringList ret;
+    int lastCapturedIndex = match.lastCapturedIndex();
+    for (int i = 1; i <= lastCapturedIndex; i++)
```

Comment:
Don't you know exactly what index you need?

Reply:
No, I won't know the index. Due to the way the regular expression matching occurs, every capturing group in the regex has a corresponding group index even when that group wasn't matched. So the regex: "([0-9]+)|([A-Z]+) has 2 groups at index 1 and 2 respectively, even though only one of them will ever be populated at once.

With the move to user supplied regexs for episode filtering, I don't know how many groups there will be at compile time. I have to iterate through all the groups, check to see if they're populated, and append it to the episode name.